w3d_vgui.lua

```diff
@@ -114,12 +125,11 @@ hook.Add("KeyPress", "w3d:KeyPress", function(pPlayer, iKey)
 
     ClearButtonsCache()
 
-    for sId, tInfos in pairs(w3d.tCache.tButtons or {}) do
+    for i, tInfos in ipairs(w3d.ReworkButtonsTable() or {}) do
```

Comment:
If you think that you're doing something more optimized, you're wrong. In fact, you're multiplying the execution time. Instead of looping one time with pairs, you're looping one time with pairs, creating a new copy of each element and adding a new value to it, then insert it into a new table with table.insert and finally loop through this new table. There's no way that is better than my previous solution.